run_bot.py

```diff
@@ -20,37 +36,44 @@ async def on_message(message):
     if not client.user.mentioned_in(message):
         return
     
-    if len(message.mentions) != 2:
+    if len(message.mentions) != 2 and len(message.mentions) != 3:
         print("mentioned count was: %s" % len(message.mentions))
         return
-
-    target = message.mentions[1]
-    if target == client.user:
-        target = message.mentions[0]
-
-    serialized_target = "chains/%s" % target.id
-
-    if not path.exists(serialized_target):
-        print("target %s for message %s doesn't exist" % (serialized_target, message))
-        return
-
-    model = None
-    with open(serialized_target) as f:
-        json = f.readlines()[0]
-        model = markovify.Text.from_json(json)
-
-    tries = 0
-    max_tries = 15
-    # sometimes the chain will generate a None message
-    while tries < max_tries:
-        sentence = model.make_sentence()
-        if sentence is not None:
-            await message.channel.send(sentence, allowed_mentions=discord.AllowedMentions(everyone=False, users=False, roles=False, replied_user=False))
+    
+    if len(message.mentions) == 2:
+        target = message.mentions[1]
+        if target == client.user:
+            target = message.mentions[0]
+        serialized_target = "chains/%s" % target.id
+        if not path.exists(serialized_target):
+            print("target %s for message %s doesn't exist" % (serialized_target, message))
             return
-        tries += 1
-
-    print("exceeded max tries")
-    return
-
+        output_message = generate_message(serialized_target)
+    else:
```

Comment:
Instead of this, use `filter` or something similar to make a list of mentions without the bot, then loop over that list.